components/webui/client/src/components/SqlEditor/index.tsx

```diff
@@ -0,0 +1,79 @@
+import {useEffect} from "react";
+
+import {
+    Editor,
+    EditorProps,
+    useMonaco,
+} from "@monaco-editor/react";
+import {language as sqlLanguage} from "monaco-editor/esm/vs/basic-languages/sql/sql.js";
+
+import "./monaco-config";
+
+
+type SqlEditorProps = Omit<EditorProps, "language">;
+
+/**
+ * Monaco editor with highlighting and autocomplete for SQL syntax.
+ *
+ * @param props
+ * @return
+ */
+const SqlEditor = (props: SqlEditorProps) => {
+    const monaco = useMonaco();
+
+
+    useEffect(() => {
+        if (null === monaco) {
+            return () => {
+            };
+        }
+
+        // Adds autocomplete suggestions for SQL keywords on editor load
+        const provider = monaco.languages.registerCompletionItemProvider("sql", {
+            provideCompletionItems: (model, position) => {
+                const word = model.getWordUntilPosition(position);
+                const range = {
+                    startLineNumber: position.lineNumber,
+                    endLineNumber: position.lineNumber,
+                    startColumn: word.startColumn,
+                    endColumn: word.endColumn,
+                };
+                const suggestions = sqlLanguage.keywords.map((keyword: string) => ({
+                    detail: "SQL Keyword",
```

Comment:
The name "Keyword" can be confusing to end users.

how about just
```suggestion
                    detail: "SQL",
```

in the future we can add UDF as well, so maybe this is better:
```suggestion
                    detail: "Presto SQL (CLP)",
```